### ISSUE TITLE
Don't set an initial speed when taking off and changing to set speed mode

### ISSUE THREAD

**Issue by ghost** [pioneerspacesim/pioneer#2251]
It's kind of annoying to take off from the first system and suddenly find yourself falling. But it's definitely more annoying to change the mode to set speed, and being pulled forwards all of a sudden.

Not sure about the proper way to fix this. The immediate problem is solved with this patch:

``` diff
diff --git a/src/ShipController.cpp b/src/ShipController.cpp
index 88e49e1..77f9595 100644
--- a/src/ShipController.cpp
+++ b/src/ShipController.cpp
@@ -303,7 +303,7 @@ void PlayerShipController::SetFlightControlState(FlightControlState s)
                m_ship->AIClearInstructions();
                //set desired velocity to current actual
                if (m_flightControlState == CONTROL_FIXSPEED) {
-                       m_setSpeed = m_setSpeedTarget ? m_ship->GetVelocityRelTo(m_setSpeedTarget).Length() : m_ship->GetVelocity().Length();
+                       m_setSpeed = m_setSpeedTarget ? m_ship->GetVelocityRelTo(m_setSpeedTarget).Length() : std::max(-(m_ship->GetVelocity()*m_ship->GetOrient()).z, 0.0);
                }
                //XXX global stuff
                Pi::onPlayerChangeFlightControlState.emit();
```

The patch sets the speed to the positive forward component of the current speed, or zero if negative. Maybe it should snap to 0 if it's below an epsilon. Maybe it should be set to negative if the current horizontal speed is negative. I think it's an improvement over what there is at the moment in any case, as it reduces the surprise factor, but there are other things to consider.

FFE keeps the current setting when changing the mode and going back to set speed mode, which is great too. Maybe that's the behaviour to aim for, initializing the current speed setting to zero when taking off. That would solve the problem too.

An alternative (or perhaps complementary) solution is to change the mode to set speed mode on takeoff, instead of manual. That has the advantage of not falling immediately after takeoff, but that's a minor annoyance once you get used to it (similar to having to remember about the undercarriage), and needing some manual operation in order to get you started is OK.

**Comment by JustinKerbice** [comment]
As pgimeno said, I am also for no automatic take off at all (or better: an option which is disabled by default: auto take-off Y/N).

As a new player, I read first tutorial, which wrongly assume we depart from a space station whereas I choose to start on Earth, going up, no thrust, didn't get any chance and time to get the right key to use, ship fall down back on the ground (quite tough it is not destroyed !).

I guess many newcomers will just try, get a "drop-crash" without understand what's just happened and just leave IMHO.

**Comment by corundscale** [comment]
I'm for no automatic takeoff and for defaulting to set speed mode when launching from surface - the latter is the only safe and sane configuration for a vehicle.

**Comment by fluffyfreak** [comment]
So you're expecting to just sit on the pad after "take-off", that's not how Pioneer works, the physics system does not allow for it, you can take-off and hover which is what I would prefer automatically happened but the physics does not allow for a resting contact.

**Comment by pgimeno** [comment]
I see your point, but landing on a planet seems to put the ship in a special static state where you have to press F5 to be vertically ejected. Can't that be done for starports?

**Comment by fluffyfreak** [comment]
It's actually the same state :)
When you take-off from a pad you're being vertically ejected too.

It's different for space stations.

This whole thing is on the neverending-todo-list-of-doom for things that need to be rewritten so it's not forgotten, other stuff has just gotten in the way.

**Comment by corundscale** [comment]
Well, the idea is to not get vertically ejected, just to get permission and eject yourself manually the way you do from rough landed state.

**Comment by pgimeno** [comment]
Yes, that would be part of the idea. When selecting "Request Launch" from the ATC, you're not immediately ejected; instead, the flag that you've requested permission is set (if there's one) and you return to the outside view. You can then press F5 to eject when you're ready.

The second part of my suggestions in the OP was to automatically change to Set Speed mode immediately after launching with F5. This part is independent of the former. It can be perhaps more controversial. I've tried with my current version (it's very old but hopefully there haven't been many changes in this area): pressing F5 twice in quick succession leaves the ship suspended in pretty much the same position as if you weren't ejected (assuming the vertical thrust of your ship in that position is enough to overcome gravity). From there you can start to manoeuver.

**Comment by corundscale** [comment]
Well, in gravity conditions it makes no sense not to start your vehicle in a mode that's best suited to deal with gravity. You don't build a vehicle specifically to be hard to control - you don't build car that speeds out of control the moment you hit the ignition and you wouldn't build a commercial spacecraft easily capable of hovering in such way that it would tend to to tumble back into the ground and turn into a fireball moments after takeoff. 

So I'm all for set speed being the default mode after surface liftoff, unless launch gets drastically reformulated to include, for example, being fired from a linear accelerator by ground control.
